.gitignore

```diff
@@ -20,14 +20,18 @@ junit-results.xml
 *.tmproj
 *.png
 *.dat
+# make sure we ship data files
+!mne/data/**/*.dat
+!mne/data/**/*.fif
+!mne/data/**/*.fif.gz
+!mne/icons/**/*.png
 .DS_Store
 events.eve
 foo-lh.label
 foo.lout
 bar.lout
 foobar.lout
```

Comment:
I think I'm somewhere in the middle on this issue --- I don't want to strip down `.gitignore` and force devs to do extra work to get back to where things were before this PR w/r/t `git status` and untracked files.

At the same time, entries like these (`foobar.lout`?) seem like things that are almost surely generated by tests and probably ought to have been cleaned up by the test (or written to a temp dir in the first place).  I'd be in favor of a targeted clean-up of entries like this. Could be in a separate PR.